Scripts/Runtime/Core/Util/QuaternionExtension.cs

```diff
@@ -0,0 +1,256 @@
+using System;
+using System.Runtime.CompilerServices;
+using Unity.Mathematics;
+using UnityEngine;
+
+namespace Anvil.Unity.Core
+{
+    /// <summary>
+    /// A collection of utilities for working with <see cref="quaternion"/>
+    /// </summary>
+    public static class QuaternionExtension
```

Comment:
Does this imply that we must have the `Unity.Mathematics` package anytime we use UnityCore? Is there a way to enforce that?

Reply:
Ah ok, if we already consider it a dependency then that's fine